src/lib/hooks/useSavedCourses.ts

```diff
@@ -173,6 +175,20 @@ export const useSavedCourses = (): SavedCourses => {
       data.map((course) => {
         if (equals(course, existingCourse)) {
           course.selected = selected;
+          if (!selected) course.showSections = false;
+          else course.showSections = true;
```

Comment:
```suggestion
          course.showSections = selected;
```
does this work? If not it could also be
```suggestion
          course.showSections = selected ? true : false;
```

Reply:
Changed to `course.showSections = selected ? true : false;`